Fix moveRegularItemOnEviction function

vinser52 opened the pull request:
@igchor Please review

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/cachelib/36)
<!-- Reviewable:end -->

vinser52 commented:
In this PR I have fixed the `moveRegularItemOnEviction` function. The fix contains the following:
1. Fixed the comment of the function which describes the return value of `moveRegularItemOnEviction` function. The function tries to move the content of the `oldItem` to the `newItemHdl` in order to recycle the `oldItem`. If the function succeeds it returns the handle to the `oldItem`, so that it can be reused for the new stuff. Otherwise, it returns an empty handle.
2. If `replaceInMMContainer(oldItemPtr, *newItemHdl)` or `newItemHdl->isAccessible()` return `false` it means some concurrent thread removed this key or called inserOrReplace, but we still can recycle the `oldItem`.
3. Removed code block related to chainedItem from `moveRegularItemOnEviction` because this function intended to work only with regular Items and we have no logic for chainedItem yet.

igchor reviewed (commented):
LGTM but it would be good to find out why similar logic is not applied in moveRegularItem.

> *[Reviewable](https://reviewable.io/reviews/pmem/cachelib/36#-:-Mv-3LSx7KvvPjUGYTsh:b-ics5co)* status: 0 of 2 files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

Commit:
Implement NUMA binding support for SysVShmSegment

Commit:
Extend cachebench to bind memory tiers to NUMA nodes

Commit:
Adding AllocatorMemoryTiersTest for NUMA bindings

Commit:
Implement NUMA binding support for PosixShmSegment

Commit:
Fix deprecation warning in multitier example

Commit:
Merge pull request #15 from vinser52/fix_deprecation_warn

Fix deprecation warn

Commit:
Update CI to use intel/CacheLib repo (#17)

Commit:
Clean-up test's blacklist (#18)

Commit:
Merge pull request #13 from vinser52/numa_binding

Initial support of NUMA bindings

Commit:
Fix ReaperSkippingSlabTraversalWhileSlabReleasing test

The issue was caused by incorrect behaviour of the
CacheAllocator<CacheTrait>::tryEvictToNextMemoryTier method in case the
evicted item is expired. We cannot simply return a handle to it, but we need
to remove it from the access container and MM container.

Commit:
Merge pull request #19 from vinser52/fix_allocator_test

Fix ReaperSkippingSlabTraversalWhileSlabReleasing test

Commit:
Fix moveRegularItemWithSync and add tests
